dbms/src/Client/InterruptListener.h

```diff
@@ -15,6 +15,40 @@ namespace ErrorCodes
 	extern const int CANNOT_UNBLOCK_SIGNAL;
 }
 
+#ifdef __APPLE__
+int sigtimedwait(const sigset_t *set, siginfo_t *info, const struct timespec *timeout) {
+        struct timespec elapsed = { 0, 0 };
+	struct timespec remaining;
+        sigset_t pending;
+        int signo;
+        const long ns = 200000000L;
+	struct timespec sleep = {0, ns};
+
+        do {
+                sigpending(&pending); /* doesn't clear pending queue */
```

Comment:
Yeah, cancelling doesn't work, because sigpedning doesn't dequeue a signal. Will think how to fix that.